react/src/components/atoms/buttons/ButtonWithIcon/index.js

```diff
@@ -21,28 +21,33 @@ class ButtonWithIcon extends React.Component {
     }
   }
   render() {
-    let classNames = this.props.classes.join(' ');
-    if (this.props.canExpand) {
+    const { classes, canExpand, expanded, capitalized, iconSize, iconColor, icon, type } = this.props
+    let classNames = classes.join(' ');
```

Comment:
Instead of joining this at this line, why not join after all the logic for classNames is done? That way you can just push new items onto the classes array instead of appending potentially empty strings.

Reply:
sure will update